kyo-prelude/shared/src/main/scala/kyo2/kernel/Boundary.scala

```diff
@@ -39,6 +39,10 @@ object Boundary:
 
     inline given derive[Ctx, S]: Boundary[Ctx, S] = ${ boundaryImpl[Ctx, S] }
 
+    def apply[Ctx, S](using boundary: Boundary[Ctx, S]): Boundary[Ctx, S] = boundary
+
+    private def create[Ctx, S]: Boundary[Ctx, S] = new Boundary(())
```

Comment:
This is needed because it's not possible to use an `AnyVal` with a private constructor in an `inline` method.